federation/deploy/deploy.sh

```diff
@@ -0,0 +1,141 @@
+readonly KUBE_ANYWHERE_FEDERATION_VERSION="v0.9.0"
+readonly KUBE_ANYWHERE_FEDERATION_CHARTS_IMAGE="gcr.io/madhusudancs-containers/federation-charts"
+readonly KUBE_ANYWHERE_FEDERATION_CHARTS_VERSION="v0.9.0"
+
+readonly GOOGLE_APPLICATION_CREDENTIALS="${GOOGLE_APPLICATION_CREDENTIALS:-${HOME}/.config/gcloud/application_default_credentials.json}"
+readonly KUBE_CONFIG_DIR="${KUBE_CONFIG_DIR:-${HOME}/.kube}"
+readonly KUBE_CONFIG="${KUBE_CONFIG:-${HOME}/.kube/config}"
+
+function pull_installer() {
+  kube::log::status "Pulling installer images"
+  docker pull "${KUBE_ANYWHERE_FEDERATION_IMAGE}:${KUBE_ANYWHERE_FEDERATION_VERSION}"
+  docker pull "${KUBE_ANYWHERE_FEDERATION_CHARTS_IMAGE}:${KUBE_ANYWHERE_FEDERATION_CHARTS_VERSION}"
+}
+
+function ensure_files() {
+  kube::log::status "Ensure provider is supported"
+  if [[ "${KUBERNETES_PROVIDER:-}" != "gce" ]]; then
+    echo "Supported providers: \"gce\""
+    exit 1
+  fi
+
+  kube::log::status "Ensure credential files exist"
+  if [[ ! -f "${GOOGLE_APPLICATION_CREDENTIALS}" ]]; then
+    echo "Please ensure Google credentials file \""${GOOGLE_APPLICATION_CREDENTIALS}"\" exists."
+    exit 1
+  fi
+
+  if [[ ! -f "${KUBE_CONFIG}" ]]; then
+    echo "Please ensure kubeconfig file \""${KUBE_CONFIG}"\" exists."
+    exit 1
+  fi
+}
+
+function kube_action() {
+  : "${FEDERATION_OUTPUT_ROOT:?must be set}"
+
+  local -r action="${1:-}"
+  kube::log::status "Action: ${action} clusters"
+  docker run \
+    --user="$(id -u):$(id -g)" \
+    -m 12G \
+    -v "${GOOGLE_APPLICATION_CREDENTIALS}:/.config/gcloud/application_default_credentials.json:ro" \
+    -v "${KUBE_CONFIG_DIR}:/.kube" \
+    -v "${FEDERATION_OUTPUT_ROOT}:/_output" \
+    "${KUBE_ANYWHERE_FEDERATION_IMAGE}:${KUBE_ANYWHERE_FEDERATION_VERSION}" \
+    "${action}"
+}
+
+function federation_action() {
+  : "${FEDERATION_OUTPUT_ROOT:?must be set}"
+
+  local -r action="${1:-}"
+  kube::log::status "Action: ${action} federation components"
+  docker run \
+    -m 12G \
+    -v "${KUBE_CONFIG}:/root/.kube/config:ro" \
+    -v "${FEDERATION_OUTPUT_ROOT}:/_output" \
+    "${KUBE_ANYWHERE_FEDERATION_CHARTS_IMAGE}:${KUBE_ANYWHERE_FEDERATION_CHARTS_VERSION}" \
+    "${action}"
+}
+
+function redeploy_federation() {
+  : "${FEDERATION_OUTPUT_ROOT:?must be set}"
+
+  local -r action="${1:-}"
+  kube::log::status "${action} federation components"
+  docker run \
+    -m 12G \
+    -v "${KUBE_CONFIG}:/root/.kube/config:ro" \
+    -v "${FEDERATION_OUTPUT_ROOT}:/_output" \
+    "${KUBE_ANYWHERE_FEDERATION_CHARTS_IMAGE}:${KUBE_ANYWHERE_FEDERATION_CHARTS_VERSION}" \
+    "${action}"
+}
+
+readonly ACTION="${1:-}"
+case "${ACTION}" in
+  "")
+  echo 'Action must be one of [init, deploy_clusters, deploy_federation, \
+    destroy_federation, destroy_clusters, redeploy_federation], \
+    got: '"${ACTION}"
+  exit 1
+  ;;
+  "init")
+  pull_installer
+  ;;
+  "deploy_clusters")
```

Comment:
What do you think about adding a "deploy" action that deploys all kube clusters and federation control plane. Same for "destroy"?
Fine to add in a follow up PR.